### DISCUSSION TITLE
Should I go Multi Agent or is there a better way?

### DISCUSSION THREAD

**Discussion by mike-adonis** [langchain4j/langchain4j#1976] [Q&A]
How do you recommend using langchain to build out a multi process chatbot ? For example I'm building a recipe dicovery chatbot that you can ask about recipes for dishes you're interested in, or you can just choose to order the ingredients if you already know what you want, alternatively you could discover a recipe by having conversations based on your preferences and likes and then proceed to ordering.

I'm approaching this using an intent analyzer that returns an enum which I use to choose the agent(with custom system message and user message, shared persistent chat memory store across all agents and specific tools for that agent) 

I'm not getting the best result with this so I wonder if there a better way to do this ?. I would like to have a smooth conversation transition from discovery to purchase and tracking while maintaining relevant context and customer preferences.

I have thought about adding the individual agents as tools to a main agent but then I would need a very polymorphic return type which is difficult to guarantee as the model gets larger and more complex

**Comment by LizeRaes** [comment]
Hi @mike-adonis if you set up a system similar to ours, just be aware that our code was a PoC. One of the things that is not correctly implemented is that I sometimes return a hardcoded message (discarding the last LLM answer) but do not reflect that change in the shared memory. For a real application, one should definitely handle that properly.
Curious to hear how it goes for you!
